### PR TITLE
fix: remove unnecessary dependency on libcst and scripts

### LINKED ISSUE #218: Remove dependency on libcst
Python Microgenerator libraries list libcst as a dependency because it is needed for the keyword fixup script. Requiring libcst causes conflicts with some AWS libraries.

Context: https://github.com/googleapis/python-bigquery/issues/302#issuecomment-704805518

Remove libcst as a dependency and edit the fixup script to raise an error if the library is not installed.

### PR DESCRIPTION
**Pull request by HemangChothani**
Fixes #218

### CODE REVIEW

**Review by tseaver — CHANGES_REQUESTED** [review]
#218 says:

> Remove libcst as a dependency and edit the fixup script to raise an error if the library is not installed.

Please restore the scripts, and edit as directed.

**Comment by HemangChothani** [comment]
@tres I found in [bigquery PR ](https://github.com/googleapis/python-bigquery/pull/308), that they deleted the file and enter in exclude in syth.py so followed the same. Please guide the correct way.

**Comment by tseaver** [comment]
@busunkim96 you specified in #218 that the fix was to remove the dependency, and update the script(s) to raise an error if `libcst` could not be imported.  Please clarify.

**Comment by HemangChothani** [comment]
@busunkim96 Could you please give some clarification?

**Comment by tswast** [comment]
> was the script not useful for bigquery by chance?

Correct. It was useless for BigQuery since we wrap the client objects in the BigQuery Storage API and in the core library we don't use the client objects at all. We only need the generator to make the compiled protos in the core library.

Still, I think it makes sense to make `libcst` an optional dependency, as it conflicts with the `awssdk` library due to incompatible `YAML` version requirements.

File: scripts/fixup_admin_v1_keywords.py [hunk 1]
```diff
@@ -17,7 +17,12 @@
 
 import argparse
 import os
-import libcst as cst
+
+try:
+    import libcst as cst
+except ImportError as exception:
+    raise ImportError(exception.__class__.__name__ + ": " + exception.msg)
```
Comment (tseaver) [inline scripts/fixup_admin_v1_keywords.py:24]: Pointless to raise another `ImportError` without extra info.  Instead, please add the message specified by @crwilcox, suggesting that the user run  `python -m pip install "libcst >= 0.2.5"`.

File: scripts/fixup_firestore_admin_v1_keywords.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 # You may obtain a copy of the License at
 #
 #     http://www.apache.org/licenses/LICENSE-2.0
-#
+#nox
```
Comment (tseaver) [inline scripts/fixup_firestore_admin_v1_keywords.py:10]: Stray line.  Please remove

File: scripts/fixup_firestore_admin_v1_keywords.py [hunk 3]
```diff
@@ -17,7 +17,12 @@
 
 import argparse
 import os
-import libcst as cst
+
+try:
+    import libcst as cst
+except ImportError as exception:
+    raise ImportError(exception.__class__.__name__ + ": " + exception.msg)
```
Comment (tseaver) [inline scripts/fixup_firestore_admin_v1_keywords.py:24]: As above here.

File: scripts/fixup_firestore_v1_keywords.py [hunk 4]
```diff
@@ -17,7 +17,12 @@
 
 import argparse
 import os
-import libcst as cst
+
+try:
+    import libcst as cst
+except ImportError as exception:
+    raise ImportError(exception.__class__.__name__ + ": " + exception.msg)
+
```
Comment (tseaver) [inline scripts/fixup_firestore_v1_keywords.py:25]: As above here.

File: setup.py [hunk 5]
```diff
@@ -80,10 +79,6 @@
     install_requires=dependencies,
     extras_require=extras,
     python_requires=">=3.6",
-    scripts=[
-        "scripts/fixup_firestore_v1_keywords.py",
-        "scripts/fixup_admin_v1_keywords.py",
-    ],
```
Comment (tseaver) [inline setup.py:86]: Remove this change, please.

File: synth.py [hunk 6]
```diff
@@ -38,7 +38,11 @@
     s.move(
         library / f"google/cloud/firestore_{version}",
         f"google/cloud/firestore_{version}",
-        excludes=[library / f"google/cloud/firestore_{version}/__init__.py"],
+        excludes=[
+            library / f"google/cloud/firestore_{version}/__init__.py",
+            "scripts/fixup_firestore_v1_keywords.py"
+        clear
+        ],
```
Comment (tseaver) [inline synth.py:45]: Remove this change, please.

File: synth.py [hunk 7]
```diff
@@ -60,7 +64,10 @@
     s.move(
         library / f"google/cloud/firestore_admin_{version}",
         f"google/cloud/firestore_admin_{version}",
-        excludes=[library / f"google/cloud/admin_{version}/__init__.py"],
+        excludes=[
+            library / f"google/cloud/admin_{version}/__init__.py",
+            "scripts/fixup_admin_v1_keywords.py"
+        ],
```
Comment (tseaver) [inline synth.py:70]: Remove this change, please.

**Review by tseaver — CHANGES_REQUESTED** [review]

File: scripts/fixup_firestore_v1_keywords.py (hunk 4 above)
Comment (HemangChothani) [reply scripts/fixup_firestore_v1_keywords.py:25]: Done, not sure why didn't reflect here.

**Review by tseaver — APPROVED** [review]

**Review by crwilcox — APPROVED** [review]

**Pull request merged**